mlir/tools/mlir-clang/Lib/clang-mlir.cc

```diff
@@ -341,39 +346,34 @@ void MLIRScanner::buildAffineLoopImpl(clang::ForStmt *fors, mlir::Location loc,
   auto affineOp = builder.create<AffineForOp>(
       loc, lb, builder.getSymbolIdentityMap(), ub,
       builder.getSymbolIdentityMap(), descr.getStep(),
-      /*iterArgs=*/llvm::None,
-      (AffineForOp::BodyBuilderFn)([&](OpBuilder &nestedBuilder,
-                                       mlir::Location loc, mlir::Value val,
-                                       ValueRange ivs) {
-        SmallVector<mlir::Value, 1> iv(ivs);
-        assert(ivs.size() == 0 && "expect single ind var");
-        if (!descr.getForwardMode()) {
-          val = nestedBuilder.create<mlir::SubIOp>(loc, val, lb);
-          val = nestedBuilder.create<mlir::SubIOp>(
-              loc,
-              nestedBuilder.create<mlir::SubIOp>(loc, ub, getConstantIndex(1)),
-              val);
-        }
-        auto idx =
-            nestedBuilder.create<mlir::IndexCastOp>(loc, val, descr.getType());
-        createAndSetAllocOp(descr.getName(), idx, 0);
+      /*iterArgs=*/llvm::None);
+
+  auto &reg = affineOp.getLoopBody();
+
+  auto val = affineOp.getInductionVar();
+
+  reg.front().clear();
 
-        auto oldpoint = builder.getInsertionPoint();
-        auto oldblock = builder.getInsertionBlock();
+  auto oldpoint = builder.getInsertionPoint();
+  auto oldblock = builder.getInsertionBlock();
+
```

Comment:
nit: I would use `OpBuilder::InsertionGuard guard(builder)` here to make sure I don't mess up with the insertion point out of the context of this function (ignore this if it is intended). You may discard those `oldpoint` and `oldblock` when using that guard.

Reply:
The way it's set up, it is intended (the call to visit will insert code at the location of the builder).